nixos/modules/services/networking/unbound.nix

```diff
@@ -139,10 +248,24 @@ in
         RestartSec = "5s";
       };
     };
-
-    # If networkmanager is enabled, ask it to interface with unbound.
-    networking.networkmanager.dns = "unbound";
-
   };
 
+  imports = [
+    (mkRenamedOptionModule [ "services" "unbound" "interfaces" ] [ "services" "unbound" "settings" "server" "interface" ])
+    (mkChangedOptionModule [ "services" "unbound" "allowedAccess" ] [ "services" "unbound" "settings" "server" "access-control" ] (
+      config: map (value: "${value} allow") (getAttrFromPath [ "services" "unbound" "allowedAccess" ] config)
+    ))
+    (mkRemovedOptionModule [ "services" "unbound" "forwardAddresses" ] ''
+      Add a new setting:
+      services.unbound.settings.forward-zone = [{
+        name = ".";
+        forward-addr = [ # Your current services.unbound.forwardAddresses ];
+      }];
+      If any of those addresses are local addresses (127.0.0.1 or ::1), you must
+      also set services.unbound.settings.server.do-not-query-localhost to false.
+    '')
+    (mkRemovedOptionModule [ "services" "unbound" "extraConfig" ] ''
```

Comment:
Is there anything meaningful you can do with `extraConfig` that can't be achieved with the new config options you have introduced?

Reply:
I could probably include `extraConfig` right after the settings clause as it was done before, but it might conflict with other clauses depending on how previous configuration was done. To avoid any surprises I think it's better to leave it as removed.

Reply:
I don't mean to give the wrong impression: having `extraConfig` is **BAD**. I just want to make sure that `settings` can cover every scenario... and if it cannot we should remedy that :+1:

Reply:
`settings` can indeed cover every scenario.

Reply:
Wonderful! :tada: